File: Cpp/49.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> sol;
        map<map<char, int>, vector<string>> m;
        for(int i = 0; i < strs.size(); i++){
            map<char, int> mm;
            for(auto el : strs[i]){
                mm[el]++;
            }
            m[mm].push_back(strs[i]);
        }
        for(auto it = m.begin(); it != m.end(); it++){
            sol.push_back(it->second);
        }
        return sol;
    }
};

File: Python/49.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        sol = []
        mm = {}
        for el in strs:
            print(el)
            m = {}
            for c in el:
                if c not in m:
                    m[c] = 1
                else:
                    m[c] = m[c] + 1
            skeys = sorted(m.keys())
            sm = {i: m[i] for i in skeys}
            print(str(sm))
            if str(sm) in mm:
                mm[str(sm)].append(el)
            else:
                mm[str(sm)] = [el]
        for k, v in mm.items():
            sol.append(v)
        return sol
